pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -426,7 +402,34 @@ func (rc *reconciler) syncStates(podsDir string) {
 
 }
 
-// Reconstruct Volume object and reconstructedVolume data structure by reading the pod's volume directories
+func (rc *reconciler) cleanupMounts(volume podVolume) {
+	if rc.desiredStateOfWorld.VolumeExistsWithSpecName(volume.podName, volume.volumeSpecName) {
+		return
+	}
+	glog.V(2).Infof("Reconciler sync states: could not find information (PID: %s) (Volume SpecName: %s) in desired state, clean up the mount points",
+		volume.podName, volume.volumeSpecName)
+	mountedVolume := operationexecutor.MountedVolume{
+		PodName:             volume.podName,
+		VolumeName:          v1.UniqueVolumeName(volume.volumeSpecName),
+		InnerVolumeSpecName: volume.volumeSpecName,
+		PluginName:          volume.pluginName,
+		PodUID:              types.UID(volume.podName),
+	}
+	volumeHandler, err := operationexecutor.NewVolumeHandlerWithMode(volume.volumeMode, rc.operationExecutor)
+	if err != nil {
+		glog.Errorf(mountedVolume.GenerateErrorDetailed(fmt.Sprintf("operationExecutor.NewVolumeHandler for UnmountVolume failed"), err).Error())
+		return
+	}
+	// TODO: Currently cleanupMounts only includes UnmountVolume operation. In the next PR, we will add
+	// to unmount both volume and device in the same routine.
+	err = volumeHandler.UnmountVolumeHandler(mountedVolume, rc.actualStateOfWorld)
```

Comment:
Is this asynchronous umount operation safe? Consider this scenario:
1. When kubelet started, we can't find the `podVolume` in desired state, so we start a independent go routine to tear down this volume
2. Then another new pod is added to use the same volume, `reconciler` discovers this pod in next cycle and starts another go routine to set up this volume
3. If the second go routine starts before the first one finished, two volume operations on the same device might happen concurrently and cause race condition because they have different unique volume names(the umount go routine uses `volumeSpecName` as unique volume name, not the real one)